lib/fastlane/actions/ipa.rb

```diff
@@ -0,0 +1,112 @@
+module Fastlane
+  module Actions
+
+    module SharedValues
+      IPA_OUTPUT_PATH = :IPA_OUTPUT_PATH
+      DSYM_OUTPUT_PATH = :DSYM_OUTPUT_PATH
+    end
+
+    # -w, --workspace WORKSPACE Workspace (.xcworkspace) file to use to build app (automatically detected in current directory) 
+    # -p, --project PROJECT Project (.xcodeproj) file to use to build app (automatically detected in current directory, overridden by --workspace option, if passed) 
+    # -c, --configuration CONFIGURATION Configuration used to build 
+    # -s, --scheme SCHEME  Scheme used to build app 
+    # --xcconfig XCCONFIG  use an extra XCCONFIG file to build the app 
+    # --xcargs XCARGS      pass additional arguments to xcodebuild when building the app. Be sure to quote multiple args. 
+    # --[no-]clean         Clean project before building 
+    # --[no-]archive       Archive project after building 
+    # -d, --destination DESTINATION Destination. Defaults to current directory 
+    # -m, --embed PROVISION Sign .ipa file with .mobileprovision 
+    # -i, --identity IDENTITY Identity to be used along with --embed 
+    # --sdk SDK            use SDK as the name or path of the base SDK when building the project 
+    # --ipa IPA            specify the name of the .ipa file to generate (including file extension) 
+
+    ARGS_MAP = {
+      workspace: '-w',
+      project: '-p',
+      configuration: '-c',
+      scheme: '-s',
+      clean: '--clean',
+      archive: '--archive',
+      destination: '-d',
+      embed: '-m',
+      identity: '-i',
+      sdk: '--sdk',
+      ipa: '--ipa'
+    }
+
+    class IpaAction
+      def self.run(params)
+        
+        # The args we will build with
+        build_args = nil
+
+        # The output directory of the IPA and dSYM
+        absolute_dest_directory = nil
+
+        # Allows for a whole variety of configurations
+        if params.first.is_a? Hash
+
+          # Used to get the final path of the IPA and dSYM
+          if dest = params.first[:destination]
+            absolute_dest_directory = Dir.glob(dest).map(&File.method(:realpath)).first
+          end
+
+          # Maps nice developer build parameters to Shenzhen args
+          build_args = params_to_build_args(params.first)
+
+        else
+          build_args = params
+        end
+
+        # If no dest directory given, default to current directory
+        absolute_dest_directory ||= Dir.pwd
+
+        if Helper.is_test?
+          Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] = File.join(absolute_dest_directory, "test.ipa")
+          Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] = File.join(absolute_dest_directory, "test.app.dSTM.zip")
```

Comment:
Wups, should be `DSYM`, right? Looks like there are no tests for the set `SharedValues` yet.